src/VecSim/types/bfloat16.h

```diff
@@ -8,17 +8,33 @@
 
 #include <cstdint>
 #include <cstring>
+#include <cmath>
+
 namespace vecsim_types {
 
 using bfloat16 = unsigned short;
 
 static inline bfloat16 float_to_bf16(const float ff) {
-    uint32_t f32 = 0;
-    memcpy(&f32, &ff, sizeof(ff));
+    uint32_t *p_f32 = (uint32_t *)&ff;
+    uint32_t f32 = *p_f32;
     uint32_t lsb = (f32 >> 16) & 1;
     uint32_t round = lsb + 0x7FFF;
     f32 += round;
     return f32 >> 16;
 }
 
+static inline float bfloat16_to_float32(bfloat16 val) {
+    float result = 0;
+    bfloat16 *p_result = (bfloat16 *)&result + 1;
+    *p_result = val;
+    return result;
+}
+
+static inline float bfloat16_to_float32_bigEndian(bfloat16 val) {
```

Comment:
consider having a `constexpr` function that checks if the target arch is big/little endian, and use it to make `bfloat16_to_float32` fit for both (by adding the `1` offset in a constexpr manner according to the function result.
This way we don't need the code duplication and the recoverable implemetation

Reply:
constexpr is checked at compilation time and we want it to be tested on runtime.
The best i can do is to unify big and little function under one templated function (similar to what i did in the normalization function)
but the test should take place at run time

Reply:
is the compiled binary compatible with other arches? I don't think so. And if not, it is something we can decide in compile time. Can we verify that?

Reply:
I pushed a fix that minimizes code duplications